spec/models/source_resolver_spec.rb

```diff
@@ -71,5 +73,37 @@
         expect(second_source.website_rank).to eq 3
       end
     end
+
+    context 'demoting top source' do
+      it 'pulls up other sources' do
+        first_source = Fabricate :source, email_rank: 1
```

Comment:
I know it seems backwards, but perhaps the order suggested above could be used here also.

`first_source = Fabricate :source, email_rank: 2`
`second_source = Fabricate :source, email_rank: 1`

Then:

`expect(first_source.reload.email_rank).to eq 1`
`expect(second_source.reload.email_rank).to eq 2`